resources/projectsyncup-design-document.md

```diff
@@ -0,0 +1,199 @@
+  * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+* Request Body: <br>
+    `{ "status": "Complete" }`
+
+### 6.8. _Track Time Spent on Task Endpoint_
+* Accepts `PUT` requests to `/tasks/:id/time`
+* Accepts a task ID and a new time spent on task, updates the ``taskTimer`` of the task with that ID.
+  * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+* Request Body: <br>
+    `{ "taskTimer": 61 }` 
+
+### 6.9. _Delete Task From Project Endpoint_
+* Accepts `DELETE` requests to `/tasks/:id`
+* Accepts a task ID, deletes the task with that ID from project.
+    * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+
+### 6.10. _View All Projects Endpoint_
+* Accepts `GET` requests to `/projects/:id`
+* Accepts a project ID, returns a list of projects created by that Manager.
+    * If the given project ID doesn't exist, will throw an ``ProjectNotFoundException``
+    * If the given Manager has not created any projects, an empty list will be returned
+
+### 6.11. _View All Assigned Tasks Endpoint_
+* Accepts `GET` requests to ` /tasks/assigned/:id`
+* Retrieves a list of all tasks assigned to given user ID.
+* Accepts a user ID, returns a list of all tasks assigned to that user.
+    * If the given user ID doesn't exist, will throw an ``UserNotFoundException``
+    * If the given user does not have any tasks, an empty list is returned.
+
+### 6.12. _View Assigned Tasks By Due Date Endpoint_
+* Accepts `GET` requests to ` /tasks/assigned/prioritized`
+* Retrieves a list of all tasks assigned to given user ID.
+  * Returns the task list in default project order (Most Recent)
+    * If the optional order parameter is provided, this API will return the task list in ``order``, reverse order, or by Due Date, based on the value of ``order``
+      * DEFAULT - same as default behavior, returns tasks in ordered assigned by Manager (Most Recent)
+      * REVERSED - returns project tasks in reversed order 
+      * DUEDATE - returns project tasks in order of due date.
+  * If the given project ID is found, but contains no tasks, the tasks list will be empty
+  * If the given project ID doesn't exist, will throw an `ProjectNotFoundException`
+
+    
+## 7. Tables
+### 7.1. `projects`
+```
+projectId // Partition Key, String
+projectName // String
+projectManager // String
+projectStatus // String
+projectTeamMembers // List<String>
+```
+
+### 7.2. `tasks`
+```
+projectId // Sort Key, String
+taskId // Partition Key, String
+taskName // String
+taskDescription // String
+assignedTo // List<String> userId-assignedTo-index partition key
+dueDate // String userId-assignedTo-index sort key
+status // String
+timeSpent // Number
+```
+
+### 7.2. `users`
+```
+userId // Partition Key, String
+userName // String
+userRole // String
+```
+## 8. Design Diagram
+* [Class Diagram](projectsyncup-class-diagram.puml)
```

Comment:
Can you include a link to a PNG of the diagram as well?  E.g. like you did for the sequence diagram on line 100.